File: gestionApp/personas/Cliente.py
from random import uniform
from tkinter import messagebox

from gestionApp.Empresa import Empresa
from gestionApp.personas.Persona import Persona
from gestionApp.excepciones import *


class Cliente(Persona):
    allClientes = []
    negociosActivos = []
    listaEventos = []
    idCounter = 0

    def __init__(self, nombre, cedula, cel, correo, cargo, activo):
        super().__init__(nombre, cedula, cel, correo)
        self.id = id
        self.cargo = cargo
        self.activo = activo
        self.probCompra = uniform(0, 0.4)
        Cliente.allClientes.append(self)
        messagebox.showinfo(message='CLIENTE creada correctamente', title='Creacion Cliente')

    def getCargo(self):
        return self.cargo

    def setCargo(self, cargo):
        self.cargo = cargo

    def getActivo(self):
        return self.activo

    def setActivo(self, activo):
        self.activo = activo

    def getProbCompra(self):
        return self.probCompra

    def setProbCompra(self, probCompra):
        self.probCompra = probCompra

    @classmethod
    def getAllClientes(cls):
        return cls.allClientes

    @classmethod
    def setAllClientes(cls, allClientes):
        cls.allClientes = allClientes

    def getProbCompra(self): return self.probCompra
    def setProbCompra(self,probCompra): self.probCompra = probCompra
    
    def getEmpresa(self): return Empresa.getNombre()
    
    
    @classmethod
    def getNegociosActivos(cls):
        return cls.negociosActivos

    @classmethod
    def setNegociosActivos(cls, negociosActivos):
        cls.negociosActivos = negociosActivos

    @classmethod
    def getListaEventos(cls):
        return cls.listaEventos

    @classmethod
    def setListaEventos(cls, listaEventos):
        cls.listaEventos = listaEventos

    @classmethod
    
    def busquedaCliente(cls, cedula):

        for i in range(len(Cliente.getAllClientes())):
            if cedula == str:
                try:
                    raise ExcepcionCamposNulos()
                except ExcepcionCamposNulos as f:
                    f.showMessage()
                return
            if Cliente.allClientes[i].cedula == cedula:
                return Cliente.getAllClientes()[i]

    @classmethod
    def eliminarCliente(cls, cedula):
       
        for i in range(len(Cliente.getAllClientes())):
            if cedula == str:
                try:
                    raise ExcepcionCamposNulos()
                except ExcepcionCamposNulos as f:
                    f.showMessage()
                return
            if (Cliente.getAllClientes(Cliente.getCedula(i))):
                return Cliente.getAllClientes(i)   
            if Cliente.allClientes[i].cedula == cedula:
                Cliente.getAllClientes().pop(i)
                      
            if Cliente.allClientes[i].cedula == cedula:
                Cliente.getAllClientes().pop(i)

        messagebox.showinfo(message='CLIENTE eliminado correctamente', title='Eliminacion Cliente')
    
    def infoCliente(self):
        return f'\t---Informacion Cliente---  \n Nombre: {self.nombre} \n Cedula: {self.cedula} \n Celular: {self.celular} \n Cargo: {self.cargo} \n Empresa: {self.getEmpresa}'

File: gestionApp/personas/Cliente.java
package gestionApp.personas;

import gestionApp.Empresa;
import gestionApp.Negocio;
import gestionApp.Evento.Evento;

import java.util.ArrayList;

public class Cliente extends Persona{
	

	private String id;
	private String cargo;
	private Boolean activo;
	private double probCompra;
	public Empresa empresa;
	private ArrayList<Negocio> negociosActivos;
	private ArrayList<Evento> listaEventos;

	public static ArrayList<Cliente> allClientes = new ArrayList<>();

	private static int idCounter = 0;
	
	public Cliente(String nombre, Empresa emp, int cedula, String cel, String correo, String cargo, Boolean activo){
		super(nombre, cedula, cel, correo);
		this.id = "0".repeat(3-String.valueOf(idCounter).length()) + String.valueOf(idCounter);
		this.cargo = cargo;
		this.activo = activo;
		this.empresa = emp;
		this.listaEventos = new ArrayList<>();
		this.probCompra = Math.random();
		idCounter += 1;
		allClientes.add(this);
		empresa.addCliente(this);
	}

	public Cliente(String nombre, int cedula, String cel, String correo, Boolean activo, String cargo){
		super(nombre, cedula, cel, correo);
		this.id = "0".repeat(3-String.valueOf(idCounter).length()) + String.valueOf(idCounter);
		this.cargo = cargo;
		this.activo = activo;
		this.listaEventos = new ArrayList<>();
		this.probCompra = Math.random();
		//empresa.addCliente(this);
		allClientes.add(this);
		idCounter += 1;

	
	}


	




	//---------------Metodos Heredados---------------
	public String getNombre(){return nombre;}
	public void setNombre(String nombre){this.nombre = nombre;}

	public int getCedula(){return cedula;}
	public void setCedula(int cedula){this.cedula = cedula;}

	public String getCel(){return this.cel;}
	public void setCel(String cel){this.cel = cel;}

	public String getCorreo(){return correo;}
	public void setCorreo(String correo){this.correo = correo;}


	//---------------Metodos Propios---------------

	public String getId(){return this.id;}

	public Empresa getEmpresa(){return empresa;}
	public void setEmpresa(Empresa empresa){this.empresa = empresa;}

	public String getCargo(){return cargo;}
	public void setCargo(String cargo){this.cargo = cargo;}

	public Boolean isActivo(){return  this.activo;}
	public void setActivo(Boolean estado){this.activo = estado;}

	public ArrayList<Negocio> getNegociosActivos(){return this.negociosActivos;}
	public void setNegociosActivos(Negocio nuevoNegocio){this.negociosActivos.add(nuevoNegocio);}

	public ArrayList<Evento> getListaEventos(){return this.listaEventos;}
	public void setAgregarEvento(Evento nuevoEvento){this.listaEventos.add(nuevoEvento);}

	public double getProbCompra() {return probCompra;}
	public void setProbCompra(double probCompra) {this.probCompra = probCompra;}

	public static ArrayList<Cliente> getAllClientes(){return allClientes;}
	public static void setAllClientes(ArrayList<Cliente> allClientes) {Cliente.allClientes = allClientes;}

	public static void clientesPotenciales() {

	}
	public static void mostrarCliente( ) {


	}

	public static void consultarClientes() {

	}
	
	public static void consultarOportunidades() {

	}
	
	public static void consultarVentas() {

	}
	
	public static void serviciosClientes() {

	}
	public String toStringEmpresa(){
		return "\n////////////////////////////\n" +
		"Empresa: " + getEmpresa().getNombre() +
		"\nNit Empresa: " + getEmpresa().getNit() +
		"\nDescripcion de la empresa a la que pertenece: \n" +
		"\n////////////////////////////\n";  
	}
	@Override
	
	public  String toString(){
		return "\tDatos del Cliente \nCliente: "+ nombre +
		 "\nIdentificado con el CC." + cedula + 
		 "\nCelular: " + cel +
		 "\nCorreo: " + correo +
		 "\nID: " + id +
		 "\nCargo: " + cargo +
		 "\nActivo en la app: " + activo +
		 "\n/////////////////////////////////////////////\n";
		 
		
	}

	



	
	
}
